src/main/java/org/folio/des/repository/criteria/CQL2JPACriteria.java

```diff
@@ -92,13 +93,47 @@ private void processSort(CQLSortNode node) throws CQLFeatureUnsupportedException
     List<Order> orders = new ArrayList<>();
     for (ModifierSet sortIndex : node.getSortIndexes()) {
       final CqlModifiers modifiers = new CqlModifiers(sortIndex);
-      orders.add(
-          CqlSort.DESCENDING.equals(modifiers.getCqlSort())
-              ? builder.desc(root.get(sortIndex.getBase()))
-              : builder.asc(root.get(sortIndex.getBase())));
+      Path<?> sortPath = root.get(sortIndex.getBase());
+      // in case of request sortby+exportTypeSpecificParameters
+      if (sortPath.getJavaType().equals(ExportTypeSpecificParameters.class)) {
+        Expression<String> innerField = JsonbNodeConverter.convertToExpression(
+          root, "jsonb.exportTypeSpecificParameters.vendorEdiOrdersExportConfig.configName", builder
+        );
+        orders.add(CqlSort.DESCENDING.equals(modifiers.getCqlSort())
+          ? builder.desc(innerField)
+          : builder.asc(innerField));
+      } else {
+        orders.add(CqlSort.DESCENDING.equals(modifiers.getCqlSort())
+          ? builder.desc(sortPath)
+          : builder.asc(sortPath));
+      }
     }
     criteria.orderBy(orders);
   }
+//  private void processSort(CQLSortNode node) throws CQLFeatureUnsupportedException {
+//    List<Order> orders = new ArrayList<>();
+//    for (ModifierSet sortIndex : node.getSortIndexes()) {
+//      final CqlModifiers modifiers = new CqlModifiers(sortIndex);
+//// sort exportTypeSpecificParameters
+//      Path<?> sortPath = root.get(sortIndex.getBase());
+//
+//      if (sortPath.getJavaType().equals(ExportTypeSpecificParameters.class)) {
+//        Expression<JsonNode> myJsonField = root.get("exportTypeSpecificParameters");
+//        Expression<String> myInnerField = builder.function("jsonb_extract_path_text", String.class, myJsonField,
+//          builder.literal("vendorEdiOrdersExportConfig"),builder.literal("configName"));
+//        orders.add(
+//          CqlSort.DESCENDING.equals(modifiers.getCqlSort())
+//            ? builder.desc(myInnerField)
+//            : builder.asc(myInnerField));
+//      } else {
+//        orders.add(
+//          CqlSort.DESCENDING.equals(modifiers.getCqlSort())
+//            ? builder.desc(sortPath)
+//            : builder.asc(sortPath));
+//      }
+//    }
+//    criteria.orderBy(orders);
+//  }
```

Comment:
```
private void processSort(CQLSortNode node) throws CQLFeatureUnsupportedException {
    List<Order> orders = new ArrayList<>();
    for (ModifierSet sortIndex : node.getSortIndexes()) {
      final CqlModifiers modifiers = new CqlModifiers(sortIndex);
// sort exportTypeSpecificParameters
      Path<?> sortPath = root.get(sortIndex.getBase());

      if (sortPath.getJavaType().equals(ExportTypeSpecificParameters.class)) {
        Expression<JsonNode> myJsonField = root.get("exportTypeSpecificParameters");
        Expression<String> myInnerField = builder.function("jsonb_extract_path_text", String.class, myJsonField,
          builder.literal("vendorEdiOrdersExportConfig"),builder.literal("configName"));
        orders.add(
          CqlSort.DESCENDING.equals(modifiers.getCqlSort())
            ? builder.desc(myInnerField)
            : builder.asc(myInnerField));
      } else {
        orders.add(
          CqlSort.DESCENDING.equals(modifiers.getCqlSort())
            ? builder.desc(sortPath)
            : builder.asc(sortPath));
      }
    }
    criteria.orderBy(orders);
  }
```
This solution itself without changing other places is also working, Which one is more convenient